### DISCUSSION TITLE
Dependencies between sidekiq 5.2.9 and 5.2.10

### DISCUSSION THREAD

**Discussion by samuelstidham** [sidekiq/sidekiq#5306] [Q&A] [answered]
I have a quick question about moving from sidekiq 5.2.9 to 5.2.10. Why is there such a big version difference for redis between these two versions? I need to upgrade to 5.2.10 to take care of some security vulnerabilities, but doing so requires me to upgrade the redis library from 3.3 to 4.5, which breaks all my redis connections. Any help related to this would be deeply appreciated.

**Comment by mperham** [comment, marked as answer]
redis 4.4 and 4.5 have a bunch of deprecations. Either I lock 5.2.10 to redis <4.5 and never update the redis gem or update 5.2.10 to use the newer APIs and lock to 4.5+. If we want security fixes, I have to keep current so that was my choice.

**Reply by samuelstidham** [reply]
Thanks for the answer! That makes sense. Any recommendations on how I can approach my code to figure out how to fix my failed connections?

**Reply by mperham** [reply]
You can always open an issue with your error messages and ask for help. Either here or in the [redis-rb project](https://github.com/redis/redis-rb).
